sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/outliers/anomalydetection/AnomalWithDataframeCrossJoin.scala

```diff
@@ -68,60 +69,27 @@ class AnomalWithDataframeCrossJoin(nTriplesRDD: RDD[Triple], objList: List[Strin
 
   // filtering triples with literal at object position
   def getObjectList(): RDD[Triple] = nTriplesRDD.filter(f => f.getObject.isLiteral())
-
   // filtering only numeric literals
-  def triplesWithNumericLit(objLit: RDD[Triple]): RDD[Triple] = objLit.filter(f => isNumeric(f.getObject.toString()))
-
-  def isNumeric(x: String): Boolean =
-    {
-      if (x.contains("^")) {
-        val c = x.indexOf('^')
-        val subject = x.substring(1, c - 1)
-
-        if (isAllDigits(subject)) true
-        else false
-      } else false
-    }
-
-  def isAllDigits(x: String): Boolean = {
-    var found = false
-    for (ch <- x) {
-      if (ch.isDigit || ch == '.') {
-        found = true
-      } else if (ch.isLetter) {
-        found = false
-      }
-    }
-
-    found
-  }
-
-  def searchedge(x: String, y: List[String]): Boolean = {
-    if (x.contains("^")) {
-      val c = x.indexOf('^')
-      val subject = x.substring(c + 2)
-      y.contains(subject)
-    } else false
-  }
+  def triplesWithNumericLit(objLit: RDD[Triple]): RDD[Triple] = objLit.filter(f => Utils.isNumeric(f.getObject.toString()))
 
   def rdfType(getHypernym: RDD[Triple]): RDD[(String, HashSet[String])] = {
 
     // filter triples with predicate as rdf:type
     val triplesWithRDFType = nTriplesRDD.filter(_.getPredicate.toString() == "http://www.w3.org/1999/02/22-rdf-syntax-ns#type")
 
-    val triplesWithDBpedia = triplesWithRDFType.filter(f => searchType(f.getObject.toString(), triplesType))
+    val triplesWithDBpedia = triplesWithRDFType.filter(f => Utils.searchType(f.getObject.toString(), triplesType))
 
     val subWithType1 = triplesWithDBpedia.map(f =>
       // ...
-      (getLocalName1(f.getSubject), (getLocalName1(f.getObject)))) // .reduceByKey(_ ++ _) //.partitionBy(new HashPartitioner(8)).persist()
+      (Utils.getLocalName(f.getSubject), (Utils.getLocalName(f.getObject)))) // .reduceByKey(_ ++ _) //.partitionBy(new HashPartitioner(8)).persist()
```

Comment:
If `f.getSubject` is a type of `Node`, it by default can call the `getLocalName` method implemented on Jena. What does it defer from the original method? consider removing this method and just call `f.getSubject.getLocalName` and the same do for other cases as well.

Reply:
same as above!